pandas/core/reshape/pivot.py

```diff
@@ -329,8 +329,7 @@ def _all_key(key):
                 piece = piece.copy()
                 try:
                     piece[all_key] = margin[key]
-                except TypeError:
-
+                except (TypeError, ValueError):
```

Comment:
do we still need the TypeError

Reply:
No, I just checked. Change has been pushed.